esp/esdllib/README.md

```diff
@@ -0,0 +1,1608 @@
+```xml
+<soap:Envelope xmlns:soap="http://schemas.xmlsoap.org/soap/envelope/">
+    <soap:Body>
+        <TestModule.TestMethodService>
+        <Context>...etc...</Context>
+        <TestMethodRequest>
+            <Row>
+                <Options>
+                    <StartDate>19760712</StartDate>
+                    <ExtendedSearchEnable>1<ExtendedSearchEnable>
+                </Options>
+                <SearchBy>
+                    <Name>
+                        <First>Jon</First>
+                        <Last>Jonson</Last>
+                    </Name>
+                </SearchBy>
+            </Row>
+        </TestMethodRequest>
+        </TestModule.TestMethodService>
+    </soap:Body>
+</soap:Envelope>
+```
+
+A script to make this change could be coded like this:
+
+```xml
+<es:BackendRequest name="TransformForRoxie" target="soap:Body/{$query}/{$request}" xmlns:es="urn:hpcc:esdl:script">
+    <es:param name="ExtendedSearchEnable" value="0"/>
+    <es:source xpath="soap:Body/*/*/Row/Options/StartDate">
+        <es:set-value target="Options/StartDate" select="Year"/>
+        <es:append-to-value target="Options/StartDate" select="Month"/>
+        <es:append-to-value target="Options/StartDate" select="Day"/>
+    </es:source>
+    <es:set-value target="Options/ExtendedSearchEnable" select={$ExtendedSearchEnable}/>
+</es:BackendRequest>
+```
+
+Let's discuss what each line of this script does.
+
+```xml
+<es:BackendRequest name="TransformForRoxie" target="soap:Body/{$query}/{$request}" xmlns:es="urn:hpcc:esdl:script">
+```
+
+The tag name of the root of the script, `BackendRequest`, specifies that it should be executed after the ESDL transformer has generated the default backend request, but just before it is sent to the backend. The action of the script can be thought of like the assignment of a variable: `x = Fn(x)`. The Left Hand Side **x** is modified by the result of the expression on the Right Hand Side **Fn(x)**. The Left Hand Side, or LHS is called the _target_ and the Right Hand Side or RHS is called the _source_. The scripting engine starts out each entry point by copying the _source_ to the _target_. In this example the _source_ and _target_ start out looking like **Sample 1**. By the time script is completed the _target_ is **Sample 2**.
+
+The `@name` attribute is used for tracing and error reporting. The `@target` attribute is an xpath specifying the default root node of the target for all subsequent operations. If it is not present the root is the `<soap:Envelope>`. The `{$query}` and `{$request}` variables are special system macros that are set to the name of the backend query and reqeust respectively.
+
+The `@xmlns` attribute is the standard XML definition of a namespace prefix. For ESDL scripts the namespace URN `urn:hpcc:esdl:script` is recommended but not required. You can also define your own prefix (we use `es`), but the namespace prefixes and definitions should be consistent across all scripts in a service.
```

Comment:
The specific URN is required. XPath extension functions are registered with this URN (hard-coded).

Not sure what to say about the prefix. We use `es` in examples. XPath extension functions are registered with `esdl` if nothing else is defined. `CustomRequestTransform` (the legacy synonym for BackendRequest) accepts both `xsdl` and `es` prefixes without explicit script definition.

Reply:
Don't add any more mentions of CustomRequestTransform.

Reply:
clarified as required